### ISSUE TITLE
glfw3.h two pairs of defines have the same value. (No big deal because they aren't related,afaik)

### ISSUE THREAD

**Issue by GraemeWilde** [glfw/glfw#791]
These defines in glfw3.h have the same values, I believe this is not how it should be, though it won't cause any issues either (afaik).

```
#define GLFW_NATIVE_CONTEXT_API     0x00036001
#define GLFW_EGL_CONTEXT_API        0x00036002

#define GLFW_ARROW_CURSOR           0x00036001
#define GLFW_IBEAM_CURSOR           0x00036002
```

**Comment by elmindreda** [comment]
Oops. Thank you for spotting this! It likely won't cause any issues, as you say, but it's still wrong.

**Comment by shurcooL** [comment]
> still wrong

Can I ask why you consider it wrong? If those defines are parts of separate namespaces, then overlap in their values is not wrong. Is it?

I agree that using different values might be a good idea because it will help catch potential user errors, where someone could mistakenly use a define for one namespace for a function that expects another, and because there is no type system to catch it, the only other way to catch such a mistake is to avoid reusing same values. Maybe this is what you meant?

**Comment by elmindreda** [comment]
@shurcooL Yes, exactly. It breaks the pattern and loses the ability to disambiguate by value.
